code/game/objects/items/wayfinding.dm

```diff
@@ -0,0 +1,77 @@
+/obj/machinery/pinpointer_dispenser
+	name = "wayfinding pinpointer dispenser"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "ticketmachine"
+	desc = "Having trouble finding your way? This machine dispenses pinpointers that point to common locations."
+	density = FALSE
+	layer = HIGH_OBJ_LAYER
+	pixel_y = 32
+	var/list/obj/item/pinpointer/wayfinding/pinpointers = list()
+
+/obj/machinery/pinpointer_dispenser/attack_hand(mob/living/carbon/user)
+	if(user.name in pinpointers)
+		to_chat(user, "<span class='warning'>There's already a pinpointer registered to [user.name]!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You take a pinpointer from [src].</span>")
+
+	var/obj/item/pinpointer/wayfinding/P = new /obj/item/pinpointer/wayfinding(get_turf(src))
+	user.put_in_hands(P)
+	P.owner = user.name
+	pinpointers[user.name] = P
```

Comment:
Thought this might be a workaround for that 1% of rounds you have two people spawn with the same name (might be more actually considering genetics and changelings)

Reply:
That would be fine for the outward facing stuff of the pinpointer, or grabbing the name from your ID and restrictkngkt to people with one. The issue is this one is added to a list you check to decide if they can have another, and two unknowns would interfere with each other.